src/Component/EventSourcing/Domain/Entity.php

```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+
+namespace Ubirak\Component\EventSourcing\Domain;
+
+abstract class Entity extends ChangesRecorder
+{
+    protected $id;
+
+    protected $parentId;
+
+    protected function __construct($id)
```

Comment:
No type hinting for $id?

Reply:
yep, for now Entity accept everything as id